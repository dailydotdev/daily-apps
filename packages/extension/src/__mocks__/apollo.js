import { createMockClient } from 'mock-apollo-client';

// eslint-disable-next-line import/prefer-default-export
export const apolloClient = createMockClient();
