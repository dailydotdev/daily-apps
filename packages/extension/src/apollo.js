import { InMemoryCache } from 'apollo-cache-inmemory';
import { CachePersistor } from 'apollo-cache-persist';
import { createPersistedQueryLink } from 'apollo-link-persisted-queries';
import { createHttpLink } from 'apollo-link-http';
import { ApolloClient } from 'apollo-client';

const cache = new InMemoryCache();

export const persistor = new CachePersistor({
  cache,
  key: 'apollo-cache-persistor',
  storage: {
    async getItem(key) {
      const res = await browser.storage.local.get(key);
      if (!(Number.isNaN(res[key]) || res[key] === undefined)) {
        return res[key];
      }
      return undefined;
    },
    setItem: (key, data) => browser.storage.local.set({ [key]: data }),
    removeItem: key => browser.storage.local.remove(key),
  },
});

const link = createPersistedQueryLink({ useGETForHashedQueries: true })
  .concat(createHttpLink({
    uri: `${process.env.VUE_APP_API_URL}/graphql`,
    useGETForQueries: true,
    credentials: 'include',
  }));

export const apolloClient = new ApolloClient({
  link,
  cache,
});
