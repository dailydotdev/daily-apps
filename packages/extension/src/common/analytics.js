import { version } from './config';
import { browserName } from './browser';
import { getCache, ANALYTICS_ID_KEY } from './cache';

const queue = [];

const initialize = async () => {
  const clientId = await getCache(ANALYTICS_ID_KEY);
  if (window.location.protocol.indexOf('moz-extension') > -1) {
    let page;

    const getMessage = (action, type, ...args) => {
      if (action !== 'send') return null;

      const prefix = `v=1&tid=${process.env.VUE_APP_GA}&cid=${clientId}&aip=1&dp=${page}`;
      if (type === 'event') {
        return `${prefix}&t=event&ec=${encodeURIComponent(args[0])}&ea=${encodeURIComponent(args[1])}&el=${encodeURIComponent(args[2])}`;
      }
      if (type === 'pageview') {
        return `${prefix}&t=pageview`;
      }
      if (type === 'timing') {
        return `${prefix}&t=timing&utc=${encodeURIComponent(args[0])}&utv=${encodeURIComponent(args[1])}&utt=${encodeURIComponent(args[2])}`;
      }
      if (type === 'exception') {
        return `${prefix}&t=exception&exd=${encodeURIComponent(args[0].exDescription)}&exf=${args[0].exFatal ? 1 : 0}`;
      }

      return null;
    };

    window.ga = (action, type, ...args) => {
      if (action === 'set') {
        if (type === 'page') {
          page = encodeURIComponent(args[0]);
        }
        return;
      }

      const request = new XMLHttpRequest();
      const message = getMessage(action, type, ...args);
      if (message) {
        request.withCredentials = true;
        request.open('POST', 'https://www.google-analytics.com/collect', true);
        request.send(message);
      }
    };

    queue.forEach(args => window.ga(...args));
  } else {
    /* eslint-disable */
    (function (i, s, o, g, r, a, m) {
      a = s.createElement(o),
        m = s.getElementsByTagName(o)[0];
      a.async = 1;
      a.src = g;
      m.parentNode.insertBefore(a, m);
    })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');
    /* eslint-enable */
    ga('create', process.env.VUE_APP_GA, { clientId });
    ga('set', 'checkProtocolTask', () => {
    });
    ga('require', 'displayfeatures');
  }
};

if (window.location.protocol.indexOf('moz-extension') > -1) {
  window.ga = (...args) => queue.push(args);
} else {
  /* eslint-disable */
  (function (i, s, o, g, r) {
    i['GoogleAnalyticsObject'] = r;
    i[r] = i[r] || function () {
      (i[r].q = i[r].q || []).push(arguments);
    }, i[r].l = 1 * new Date();
  })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');
  /* eslint-enable */
}

export default function (consent) {
  if (consent) {
    return initialize();
  }

  window.ga = () => {
  };
  return Promise.resolve();
}

export const trackPageView = (page) => {
  const prefix = browserName ? '/extension' : '/';
  const suffix = browserName ? `v=${version}&b=${browserName}` : `v=${version}`;
  ga('set', 'page', `${prefix}${page}?${suffix}`);
  ga('send', 'pageview');
};
