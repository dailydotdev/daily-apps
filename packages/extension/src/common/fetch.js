// eslint-disable-next-line import/prefer-default-export
export const fetchTimeout = (url, ms, { signal, ...options } = {}) => {
  const controller = new AbortController();
  const promise = fetch(url, { signal: controller.signal, ...options });
  if (signal) {
    signal.addEventListener('abort', () => controller.abort());
  }
  const timeout = setTimeout(() => controller.abort(), ms);
  return promise.finally(() => clearTimeout(timeout));
};
