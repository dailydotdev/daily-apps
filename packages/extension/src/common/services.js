import {
  AuthServiceImpl, ContentServiceImpl, MonetizationServiceImpl, ProfileServiceImpl,
} from '@daily/services';

const app = 'extension';
// eslint-disable-next-line import/prefer-default-export
export const contentService = new ContentServiceImpl(process.env.VUE_APP_API_URL, 30, app);
export const monetizationService = new MonetizationServiceImpl(process.env.VUE_APP_API_URL, app);
export const profileService = new ProfileServiceImpl(process.env.VUE_APP_API_URL, app);
export const authService = new AuthServiceImpl(process.env.VUE_APP_API_URL, app);
