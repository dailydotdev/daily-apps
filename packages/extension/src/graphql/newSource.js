import gql from 'graphql-tag';

export const SOURCE_BY_FEED_QUERY = gql`
  query SourceByFeed($data: String!) {
    sourceByFeed(feed: $data) {
      id, name, image, public
    }
  }`;

export const ADD_PRIVATE_SOURCE_MUTATION = gql`
  mutation AddPrivateSource($data: AddPrivateSourceInput!) {
    addPrivateSource(data: $data) {
      id, name, image, public
    }
  }`;
