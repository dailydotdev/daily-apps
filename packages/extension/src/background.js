import { authService } from './common/services';
import {
  setCache,
  getCache,
  ANALYTICS_ID_KEY,
  DATA_VERSION_KEY,
  CURRENT_DATA_VERSION,
  STATE_KEY,
} from './common/cache';

const cacheUserId = async () => {
  const profile = await authService.getUserProfile();
  return setCache(ANALYTICS_ID_KEY, profile.id);
};

const migrateVersion1 = async () => {
  const [bookmarks, pubs, settings, profile, notificationTime] = await Promise.all([
    getCache('bookmarks', []),
    getCache('publications', {}),
    getCache('settings', {}),
    getCache('profile', null),
    getCache('notificationTime', null),
  ]);
  return {
    feed: {
      bookmarks,
      publications: Object.keys(pubs).map(id => ({ id, enabled: pubs[id] })),
    },
    ui: {
      lastNotificationTime: notificationTime,
      ...settings,
    },
    user: { profile },
  };
};

const migrateCache = async () => {
  const version = await getCache(DATA_VERSION_KEY, '1');
  if (version !== CURRENT_DATA_VERSION) {
    const newCache = await migrateVersion1();
    await setCache(STATE_KEY, newCache);
  }
  await setCache(DATA_VERSION_KEY, CURRENT_DATA_VERSION);
};

browser.browserAction.onClicked.addListener(() => {
  const url = browser.extension.getURL('index.html?source=button');
  browser.tabs.create({ url, active: true });
});

browser.alarms.create('cacheUserId', { periodInMinutes: 60 });

browser.alarms.onAlarm.addListener(async (alarm) => {
  if (alarm.name === 'cacheUserId') {
    // TODO: handle error
    await cacheUserId();
  }
});

browser.runtime.onInstalled.addListener(async () => {
  await Promise.all([
    cacheUserId(),
    migrateCache(),
  ]);
});
