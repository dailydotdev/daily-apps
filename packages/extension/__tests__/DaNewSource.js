import {mount, createLocalVue} from '@vue/test-utils';
import Vuex from 'vuex';
import VueApollo from 'vue-apollo';
import icons from '@daily/components/src/icons';
import DaModal from '@daily/components/src/components/DaModal.vue';
import DaTextField from '@daily/components/src/components/DaTextField.vue';
import DaSpinner from '@daily/components/src/components/DaSpinner.vue';
import {apolloClient} from '../src/apollo';
import {fetchTimeout} from '../src/common/fetch';
import {contentService} from '../src/common/services';
import DaNewSource from '../src/components/DaNewSource.vue';
import {SOURCE_BY_FEED_QUERY, ADD_PRIVATE_SOURCE_MUTATION} from '../src/graphql/newSource';

jest.mock('../src/apollo');
jest.mock('../src/common/fetch', () => ({
  fetchTimeout: jest.fn(),
}));
jest.mock('../src/common/services', () => ({
  contentService: {
    requestPublication: jest.fn(),
  },
}));

let localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueApollo);
localVue.use(icons);
localVue.component('da-modal', DaModal);
localVue.component('da-text-field', DaTextField);
localVue.component('da-spinner', DaSpinner);

const sourceByFeedHandler = jest.fn();
const addPrivateSourceHandler = jest.fn();
apolloClient.setRequestHandler(SOURCE_BY_FEED_QUERY, sourceByFeedHandler);
apolloClient.setRequestHandler(ADD_PRIVATE_SOURCE_MUTATION, addPrivateSourceHandler);

let user;
let ui;
let store;

beforeEach(() => {
  window.ga = () => {
  };

  sourceByFeedHandler.mockReset();
  sourceByFeedHandler.mockResolvedValue({data: {sourceByFeed: null}});
  addPrivateSourceHandler.mockReset();
  addPrivateSourceHandler.mockResolvedValue({
    data: {
      addPrivateSource: {
        id: 'id',
        name: 'Name',
        image: 'https://image.com',
        public: false,
      },
    },
  });

  user = {
    namespaced: true,
    state: {
      profile: {premium: true},
    },
    getters: {
      isPremium: state => !!state.profile && !!state.profile.premium,
    },
  };

  ui = {
    namespaced: true,
    state: {},
    mutations: {
      setShowPremium: jest.fn(),
    },
  };

  store = new Vuex.Store({
    modules: {user, ui},
  });
});

it('should emit close on close button click', () => {
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  wrapper.find('.modal__close-btn').trigger('click');
  expect(wrapper.emitted().close[0]).toBeTruthy();
});

it('should autofocus on input', () => {
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  expect(wrapper.find('input').element.autofocus).toEqual(true);
});

it('should disable confirm button', () => {
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  expect(wrapper.find('button[type="submit"]').element.disabled).toEqual(true);
});

it('should enable confirm button when input is valid', () => {
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  const input = wrapper.find('input');
  input.element.value = 'https://example.com';
  input.trigger('input');
  expect(wrapper.find('button[type="submit"]').element.disabled).toEqual(false);
});

it('should show error when cannot find rss', (done) => {
  fetchTimeout.mockResolvedValue({
    json: jest.fn().mockResolvedValue({
      type: 'website',
      name: 'Example',
      logo: 'https://example.com/logo.png',
      rss: [],
    }),
  });
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  const input = wrapper.find('input');
  input.element.value = 'https://example.com';
  input.trigger('input');
  wrapper.find('button[type="submit"]').trigger('click');
  setTimeout(() => {
    expect(wrapper.find('.new-source__status').element).toMatchSnapshot();
    done();
  });
});

it('should show error when cannot scrape website', (done) => {
  fetchTimeout.mockResolvedValue({json: jest.fn().mockResolvedValue({type: 'unavailable'})});
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  const input = wrapper.find('input');
  input.element.value = 'https://example.com';
  input.trigger('input');
  wrapper.find('button[type="submit"]').trigger('click');
  setTimeout(() => {
    expect(wrapper.find('.new-source__status').element).toMatchSnapshot();
    done();
  });
});

it('should list scraped rss feeds', (done) => {
  fetchTimeout.mockResolvedValue({
    json: jest.fn().mockResolvedValue({
      type: 'website',
      name: 'Example',
      logo: 'https://example.com/logo.png',
      rss: [{title: 'A', url: 'https://a.com/feed'}, {title: 'B', url: 'https://b.com/feed'}],
    }),
  });
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  const input = wrapper.find('input');
  input.element.value = 'https://example.com';
  input.trigger('input');
  wrapper.find('button[type="submit"]').trigger('click');
  setTimeout(() => {
    expect(wrapper.find('.new-source__status').element).toMatchSnapshot();
    expect(wrapper.find('.new-source__footer').element).toMatchSnapshot();
    done();
  });
});

it('should proceed directly to source info when one rss is available', (done) => {
  fetchTimeout.mockResolvedValue({
    json: jest.fn().mockResolvedValue({
      type: 'website',
      name: 'Example',
      logo: 'https://example.com/logo.png',
      rss: [{title: 'A', url: 'https://a.com/feed'}],
    }),
  });
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  const input = wrapper.find('input');
  input.element.value = 'https://example.com';
  input.trigger('input');
  wrapper.find('button[type="submit"]').trigger('click');
  setTimeout(() => {
    expect(wrapper.find('.new-source__status').element).toMatchSnapshot();
    expect(wrapper.find('.new-source__footer').element).toMatchSnapshot();
    done();
  });
});

it('should show source exists message', async () => {
  fetchTimeout.mockResolvedValue({
    json: jest.fn().mockResolvedValue({
      type: 'website',
      name: 'Example',
      logo: 'https://example.com/logo.png',
      rss: [{title: 'A', url: 'https://a.com/feed'}],
    }),
  });
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  wrapper.setData({
    existsSource: {
      id: 'source',
      name: 'Source',
      image: 'https://source.com/logo.png',
      public: true,
    },
  });
  await wrapper.vm.$nextTick();
  expect(wrapper.find('.new-source__status').element).toMatchSnapshot();
  expect(wrapper.find('.new-source__footer').element).toMatchSnapshot();
});

it('should proceed to source info after selecting rss', async (done) => {
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  wrapper.setData({
    source: {
      type: 'website',
      name: 'Example',
      logo: 'https://example.com/logo.png',
      rss: [{title: 'A', url: 'https://a.com/feed', enabled: true}, {
        title: 'B',
        url: 'https://b.com/feed',
      }],
    },
  });
  wrapper.setData({scraped: true});
  await wrapper.vm.$nextTick();
  wrapper.find('.new-source__confirm').trigger('click');
  setTimeout(() => {
    expect(wrapper.find('.new-source__status').element).toMatchSnapshot();
    expect(wrapper.find('.new-source__footer').element).toMatchSnapshot();
    done();
  });
});

it('should request source', async (done) => {
  contentService.requestPublication.mockResolvedValue();
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  wrapper.setData({
    source: {
      type: 'website',
      name: 'Example',
      logo: 'https://example.com/logo.png',
      rss: [{title: 'A', url: 'https://a.com/feed', enabled: true}, {
        title: 'B',
        url: 'https://b.com/feed',
      }],
    },
  });
  wrapper.setData({scraped: true, validName: true, hasSelectedRSS: true});
  await wrapper.vm.$nextTick();
  wrapper.findAll('.new-source__submit button').at(0).trigger('click');
  await wrapper.vm.$nextTick();
  setTimeout(() => {
    expect(contentService.requestPublication).toBeCalledWith('https://a.com/feed');
    expect(wrapper.emitted().close[0]).toBeTruthy();
    expect(wrapper.emitted()['requested-source'][0]).toBeTruthy();
    done();
  }, 100);
});

it('should show premium popup', async (done) => {
  user.state.profile.premium = false;
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  wrapper.setData({
    source: {
      type: 'website',
      name: 'Example',
      logo: 'https://example.com/logo.png',
      rss: [{title: 'A', url: 'https://a.com/feed', enabled: true}, {
        title: 'B',
        url: 'https://b.com/feed',
      }],
    },
  });
  wrapper.setData({scraped: true, validName: true, hasSelectedRSS: true});
  await wrapper.vm.$nextTick();
  wrapper.findAll('.new-source__submit button').at(1).trigger('click');
  await wrapper.vm.$nextTick();
  setTimeout(() => {
    expect(ui.mutations.setShowPremium).toBeCalledWith(expect.anything(), true);
    expect(wrapper.emitted().close[0]).toBeTruthy();
    done();
  }, 100);
});

it('should add private source', async () => {
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  wrapper.setData({
    source: {
      type: 'website',
      name: 'Example',
      website: 'https://daily.dev',
      logo: 'https://example.com/logo.png',
      rss: [{title: 'A', url: 'https://a.com/feed'}, {title: 'B', url: 'https://b.com/feed'}],
    },
  });
  wrapper.setData({scraped: true, validName: true, hasSelectedRSS: true});
  await wrapper.vm.$nextTick();
  wrapper.findAll('.new-source__submit button').at(1).trigger('click');
  await wrapper.vm.$nextTick();
  expect(addPrivateSourceHandler).toBeCalledWith({
    data: {
      name: 'Example',
      website: 'https://daily.dev',
      image: 'https://example.com/logo.png',
      rss: 'https://a.com/feed',
    },
  });
  await wrapper.vm.$nextTick();
  expect(wrapper.find('.new-source__header').element).toMatchSnapshot();
});

it('should add private source wit manually typed name', async () => {
  const wrapper = mount(DaNewSource, {
    store,
    localVue,
    apolloProvider: new VueApollo({defaultClient: apolloClient}),
  });
  wrapper.setData({
    source: {
      type: 'website',
      name: 'Example',
      website: 'https://daily.dev',
      logo: 'https://example.com/logo.png',
      rss: [{title: 'A', url: 'https://a.com/feed'}, {title: 'B', url: 'https://b.com/feed'}],
    },
  });
  wrapper.setData({scraped: true, validName: true, hasSelectedRSS: true});
  await wrapper.vm.$nextTick();
  const input = wrapper.find('.new-source__name input');
  input.element.value = 'My Source';
  input.trigger('input');
  wrapper.findAll('.new-source__submit button').at(1).trigger('click');
  await wrapper.vm.$nextTick();
  expect(addPrivateSourceHandler).toBeCalledWith({
    data: {
      name: 'My Source',
      website: 'https://daily.dev',
      image: 'https://example.com/logo.png',
      rss: 'https://a.com/feed',
    },
  });
  await wrapper.vm.$nextTick();
  expect(wrapper.find('.new-source__header').element).toMatchSnapshot();
});
