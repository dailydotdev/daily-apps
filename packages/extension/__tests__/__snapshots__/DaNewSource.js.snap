// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add private source 1`] = `
<div
  class="new-source__header"
>
  <h3>
    Source added ðŸŽ‰
  </h3>
   
  <p>
    
      From now on new articles from this source will be added to your feed. Enjoy
    
  </p>
</div>
`;

exports[`should add private source wit manually typed name 1`] = `
<div
  class="new-source__header"
>
  <h3>
    Source added ðŸŽ‰
  </h3>
   
  <p>
    
      From now on new articles from this source will be added to your feed. Enjoy
    
  </p>
</div>
`;

exports[`should list scraped rss feeds 1`] = `
<div
  class="new-source__status"
>
  <span>
    
        2 RSS feeds found
      
  </span>
</div>
`;

exports[`should list scraped rss feeds 2`] = `
<div
  class="new-source__footer no-padding"
>
  <div
    class="radio"
  >
    <label
      class="radio-item checked"
    >
      <input
        name="rss"
        type="radio"
      />
       
      <span
        class="radio-item__checkmark"
      >
        <span
          class="radio-item__checkmark__focus"
        />
      </span>
       A
  
    </label>
    <label
      class="radio-item"
    >
      <input
        name="rss"
        type="radio"
      />
       
      <span
        class="radio-item__checkmark"
      >
        <span
          class="radio-item__checkmark__focus"
        />
      </span>
       B
  
    </label>
  </div>
   
  <button
    class="btn btn-big btn-invert new-source__confirm"
  >
    Continue
  </button>
</div>
`;

exports[`should proceed directly to source info when one rss is available 1`] = `
<div
  class="new-source__status"
>
  <span>
    
        Please confirm the details below
      
  </span>
</div>
`;

exports[`should proceed directly to source info when one rss is available 2`] = `
<div
  class="new-source__footer"
>
  <div
    class="new-source__name"
  >
    <img
      alt="Source logo"
      src="https://example.com/logo.png"
    />
     
    <div
      class="text-field"
    >
      <div
        class="text-field__box field active"
      >
        <!---->
         
        <div
          class="text-field__input-container"
        >
          <label
            class="text-field__label micro2"
            style="display: none;"
          >
            Source name
          </label>
           
          <input
            aria-label="Source name"
            autofocus="autofocus"
            class="text-field__input"
            maxlength="50"
            placeholder="Source name"
            required="required"
            type="text"
          />
        </div>
         
        <span
          class="text-field__count lil1"
        >
          
      43
    
        </span>
         
      </div>
       
      <div
        class="text-field__hint micro2"
      >
        
    
  
      </div>
    </div>
  </div>
   
  <div
    class="new-source__submit"
  >
    <div>
      <div
        class="new-source__submit__info"
      >
        <h3>
          Public
        </h3>
         
        <p>
          
            We will review your request to make sure it qualifies the standards.
          <br />
          <br />
          
            If the source is approved, it will be added to the public sources list.
            We will update you by email.
          <br />
          <br />
          
            Read the 
          <a
            href="#"
            target="_blank"
          >
            content guidelines
          </a>
        </p>
      </div>
       
      <button
        class="btn btn-big btn-hollow"
      >
        
          Request
        
      </button>
    </div>
     
    <div>
      <div
        class="new-source__submit__info"
      >
        <h3>
          Private
        </h3>
         
        <p>
          
            Personalize your feed with  custom private sources of your choice.
          <br />
          <br />
          
            This source will be added to your feed 
          <strong>
            immediately
          </strong>
          .
          
        </p>
      </div>
       
      <button
        class="btn btn-big"
      >
        
          Add now
        
      </button>
    </div>
  </div>
</div>
`;

exports[`should proceed to source info after selecting rss 1`] = `
<div
  class="new-source__status"
>
  <span>
    
        Please confirm the details below
      
  </span>
</div>
`;

exports[`should proceed to source info after selecting rss 2`] = `
<div
  class="new-source__footer"
>
  <div
    class="new-source__name"
  >
    <img
      alt="Source logo"
      src="https://example.com/logo.png"
    />
     
    <div
      class="text-field"
    >
      <div
        class="text-field__box field active"
      >
        <!---->
         
        <div
          class="text-field__input-container"
        >
          <label
            class="text-field__label micro2"
            style="display: none;"
          >
            Source name
          </label>
           
          <input
            aria-label="Source name"
            autofocus="autofocus"
            class="text-field__input"
            maxlength="50"
            placeholder="Source name"
            required="required"
            type="text"
          />
        </div>
         
        <span
          class="text-field__count lil1"
        >
          
      43
    
        </span>
         
      </div>
       
      <div
        class="text-field__hint micro2"
      >
        
    
  
      </div>
    </div>
  </div>
   
  <div
    class="new-source__submit"
  >
    <div>
      <div
        class="new-source__submit__info"
      >
        <h3>
          Public
        </h3>
         
        <p>
          
            We will review your request to make sure it qualifies the standards.
          <br />
          <br />
          
            If the source is approved, it will be added to the public sources list.
            We will update you by email.
          <br />
          <br />
          
            Read the 
          <a
            href="#"
            target="_blank"
          >
            content guidelines
          </a>
        </p>
      </div>
       
      <button
        class="btn btn-big btn-hollow"
      >
        
          Request
        
      </button>
    </div>
     
    <div>
      <div
        class="new-source__submit__info"
      >
        <h3>
          Private
        </h3>
         
        <p>
          
            Personalize your feed with  custom private sources of your choice.
          <br />
          <br />
          
            This source will be added to your feed 
          <strong>
            immediately
          </strong>
          .
          
        </p>
      </div>
       
      <button
        class="btn btn-big"
      >
        
          Add now
        
      </button>
    </div>
  </div>
</div>
`;

exports[`should show error when cannot find rss 1`] = `
<div
  class="new-source__status"
>
  <div
    class="new-source__contact error"
  >
    <span>
      Could not find RSS feed
    </span>
     
    <a
      class="btn btn-hollow"
      href="mailto:hi@daily.dev?subject=Failed to add new source"
      target="_blank"
    >
      Contact
    </a>
  </div>
</div>
`;

exports[`should show error when cannot scrape website 1`] = `
<div
  class="new-source__status"
>
  <div
    class="new-source__contact error"
  >
    <span>
      Failed to fetch information
    </span>
     
    <a
      class="btn btn-hollow"
      href="mailto:hi@daily.dev?subject=Failed to add new source"
      target="_blank"
    >
      Contact
    </a>
  </div>
</div>
`;

exports[`should show source exists message 1`] = `
<div
  class="new-source__status"
>
  <div
    class="new-source__exists"
  >
    <img
      alt="Source logo"
      src="https://source.com/logo.png"
    />
     
    <span
      class="micro1"
    >
      Source
    </span>
     
    <span
      class="micro2"
    >
      Already exists
    </span>
  </div>
</div>
`;

exports[`should show source exists message 2`] = `undefined`;
